### ISSUE TITLE
[Solved] Lag Spike every 2-3 seconds. Tried solutions proposed in this github, nothing worked [3-4 minute read]

### ISSUE THREAD

**Issue by Kaytrox** [moonlight-stream/moonlight-android#480]
Hey there,

this may have an obvious solution, I know, sometimes I am blind and am just not seeing the solution, so sorry if that's the case here.
And honestly I read through a lot of issues and tried some things, but nothing worked. 
(Specs at the bottom of post)

First, this is how my Problem looks like: https://youtu.be/oCYzPvFqS6s (Direct capture through Game Launcher, this is what I see on my phone <|> no sound because I muted everything on my phone)

What I tried after reading some issue posts on this topic:
1. Set Lan to 100Mbps Full Duplex
![image](https://user-images.githubusercontent.com/24830310/33130583-2221a10a-cf94-11e7-9ab5-66424b75ccdb.png)
2. Turned off h.265 on my phone
![image](https://user-images.githubusercontent.com/24830310/33131088-c9e48000-cf95-11e7-8781-0c949c1bd15d.png)
3. Downloaded and used the Microstutter Fix v5.2.1 (Used this for the clip and the task-manager screenshot)
https://github.com/moonlight-stream/moonlight-android/issues/349#issuecomment-307426527
4. Tried streaming in 720p30fps with 3 mbps, (no difference to 1080p30fps with 50mbps)
5. Turned off Bluetooth

I'm not using any gamepad or similar thing, this is meant to be combined with Tridef 3D and a headmount to be become a DIY and low-cost VR alternative.

I don't know what to do now, I am fairly confident that my setup should be powerful enough, as this is how my task-manager looks like while running circles (same as in the clip)
![image](https://user-images.githubusercontent.com/24830310/33131893-42915328-cf98-11e7-9da9-adeb600fb6ac.png)
[Ethernet1->VPN | Ethernet2->Phone Tethering | Ethernet3->another VPN | Ethernet4->actual connection]

And yes, if you guessed that these spikes on the cpu sync with the lagg I see on my phone, then you sir guessed correctly.
But that makes me think that this isn't a moonlight issue, so sorry if it isn't.
Still, if someone knows what the problem is, please tell me, I currently can't think of any more solutions or problem sources.

This is my wireless configuration:
![image](https://user-images.githubusercontent.com/24830310/33130935-46a72c60-cf95-11e7-9784-205ecf887ba9.png)
(Bottom one is Bandwidth and "Kanal" means channel)
Phone, PC and Router/Modem are all within a 40cm radius.
PC is connected via cable.

My phone is a Samsung Galaxy S8 (SM-G950F to be exact)
My PC:
CPU: Intel Xeon E3-1231 v3 (Equal to i7-4790)
GPU: Gtx 970
Ram: 16Gb

**Comment by Cur** [comment]
Hey!

I had this issue on my Lenovo Tab 3 10 Business.

It's caused by my device lagging.

I went thru and closed every single app - including disabling them so NOTHING but the android system it was still taking up 90% of the resources.

90% of the time after killing everything that was open - smooth as after that.

I got a new tablet (Lenovo P8) yesterday as i could not root the first one and the mouse issue was driving me insane.   Much faster specs and 0 stuttering / freezing issues.

I understand your device is a S8 which should be even more powerful than my device.... so....  how many apps do u have open at once and if you kill everything then launch moonlight does it fix the stuttering?

**Comment by Kaytrox** [comment]
Hi,

thanks for your help, really appreciate it!
Unfortunately this isn't the issue, I usually close every app and frequently run the self-clean function of my phone.
So I tested your idea by restarting the phone, activating performance mode and starting Moonlight through the Game Launcher, where I also enable Game-Performance mode.
Sadly, nothing changed.

BUT!
Retrying and testing your solution gave me an idea, and I have found the problem.
My Router just is not capable to keep a constant stream of data, no matter how much bandwidth it needs.
I noticed something like this before in games like battlefield 1, were I was constantly getting all 3 network warnings. I did not notice any lag and had a ping of usually 21-24ms. So I assumed it was my ISP (I live in a more rural area) and did not think about it affecting local connections.

So you might ask how I tested/fixed this, I simply disable W-Lan on my phone, connected my phone to my pc via USB3.0(PC) and USB-C(Phone), activated USB-Tethering in my phones settings. This way I should have a constant lagg/packetloss-free connection until I reach the hardware limits of my phone or pc.

And yes, that worked. 1080p60fps with 100mbps without additional input-lag is absolutely no problem, there are minimal microstutters, but that might just be my gpu not being able to keep a constant 60fps. Still. Its a billion times better now.

Hope this helps some people, now they at least have something to test for...


And thanks to the devs making something like this possible in the first place!
